### ISSUE TITLE
Widgets page: don't use icon-only control for no reason

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#24939]
**Describe the bug**

For accessibility, user interface controls that use only icons and no visible text are an anti-pattern, for many reasons. This issue has been discussed at length over 3 years of the Gutenberg development (so far) and the accessibility team always strongly recommended to prefer controls with visible text, where possible.

In some cases, the UI doesn't provide enough space to display text, for example in the various block editor toolbars. In these cases, the accessibility team accepted the usage of icon-only controls _as a compromise_ and to not block development. The problem is still pending a solution and some important exploration for alternative layouts were made, see for example #10524.

However, when the UI does provide sufficient space to use text, there's really no reason to use only icons.

In the new widgets page, the full-width "Add block" button displayed at the bottom of each widgets area uses only an icon:

<img width="781" alt="widgets avoid icon only when no need" src="https://user-images.githubusercontent.com/1682452/91732681-d9d6fd80-eba8-11ea-95f2-ecc1a5f4096a.png">

In this case, there's no reason to use an icon. Visible text should be used instead.

**Comment by mapk** [comment]
Good catch! We've actually updated this design since. Here's the new design which looks just like the current Block Editor.

![Screen Shot 2020-09-01 at 2 26 43 PM](https://user-images.githubusercontent.com/617986/91907812-5f1ce980-ec5f-11ea-9e4d-d07755cbdd3d.png)

So I'll close this issue regarding the Widget Editor. Thanks!

**Comment by mapk** [comment]
In case I wasn't too clear, I mean that the design has been updated, but it still has yet to be implemented in the code.

**Comment by afercia** [comment]
Thanks @mapk. I'd like to close this issue when the code will be implemented, also because the screenshot you attached in your comment shows an _empty_ widgets area while my issue is about the "add" button displayed when a widgets area does have active widgets. 

What the new design proposes in this case? Where I can see the new proposed design? Thanks.

**Comment by mapk** [comment]
>I'd like to close this issue when the code will be implemented,

Totally cool, thanks for reopening.

>my issue is about the "add" button displayed when a widgets area does have active widgets.

Maybe I wasn't clear, but these areas will act just like the Post Editor. So even with blocks, there won't be a large `+` button at the bottom.

This gives a pretty decent understanding: https://github.com/WordPress/gutenberg/issues/24663#issuecomment-682248494

Basically there's no indicator to add a block below just like in the current Editor. Adding a new block can be achieved by tapping `Enter` on the keyboard to get the `/` inserter or moving the mouse to the topbar `+` Inserter.

**Comment by afercia** [comment]
Thanks for the clarification.

> Basically there's no indicator to add a block below just like in the current Editor. 

Actually, in the block editor that depends on what the selected block type is. Paragraphs don't show any control to add a block after them. Other block types do, for example the Image block.

**Comment by mapk** [comment]
>Paragraphs don't show any control to add a block after them. Other block types do, for example the Image block.

Yep, thanks for pointing that out. It work just like that.

**Comment by afercia** [comment]
On a side note: while I do understand some of the reasoning behind that design decision, I think the fact paragraphs and other block types behave differently is confusing for users and the current behavior tends to be a bit too _smart_ to be actually useful. I'd like to see that behavior reconsidered.

**Comment by draganescu** [comment]
I believe #25635 closed this?

**Comment by afercia** [comment]
Well, the "default block list appender" is an icon-only control...

**Comment by mapk** [comment]
I suggest we close this because we're attempting to bring the widget areas on par with the post editor to maintain consistency and predictability for users.

If there's a problem with that icon because it's implemented differently than in the post editor, let's fix that, but otherwise this issue should address the overall implementation and shouldn't focus solely on the Widget Screen.

**Comment by afercia** [comment]
> we're attempting to bring the widget areas on par with the post editor to maintain consistency and predictability for users

I'm not sure this is a value _per se_. The two user interfaces are pretty different and have different needs. The block editor one is way more complex and, to some extent, I understand the need to decrease the overall visual "weight". Granted, changing the icon-only button also on the block editor would be very appreciated.

Instead, the widgets user interface is way "lighter" than the block editor one. Also, there's a lot of space to expand the icon-only button and change it to a button with visible text. 

From an accessibility perspective, icon-only controls are an anti-pattern and should be only used when there's no other option. In this widgets screen I don't see any need to use icon-only controls and I'd like to see some exploration to use buttons with visible text instead. 🙂

**Comment by draganescu** [comment]
Hi @afercia it's hard to follow what this issue in particular is about. Can we open separate issues on generic design feedback and close this one as the actual thing (the icon only add button) is now gone and the behavior is consistent with other places in WordPress. I am not saying anything about any other comment here, just a ping to form new discussion on the unrelated points on this issue.

**Comment by noisysocks** [comment]
Both the post editor and widgets editor use the same component for the block appender. A single code change will fix it in both places. I'm going to close this issue in favour of https://github.com/WordPress/gutenberg/issues/15311 which tracks the broader issue here. I've added a note to https://github.com/WordPress/gutenberg/issues/15311.

**Comment by afercia** [comment]
Fair enough, as long as the change is included in the things to have for the WordPress 5.6 release, as the Widgets page is a new feature.

**Comment by draganescu** [comment]
Howdy @afercia I really did not mean to come across as if I have an issue with anything you said :D It wasn't even feedback! 

What I meant was that as a comment here to an issue that is solved (the big plus button in the widgets editor) the real problem is less visible and we should open one or update the root issue discussing the general problem.

I think pointing this to #15311 is a good idea!
